kornia/geometry/epipolar/fundamental.py

```diff
@@ -70,9 +70,212 @@ def normalize_transformation(M: Tensor, eps: float = 1e-8) -> Tensor:
+
+    if Q == 0:
+        if R == 0:
+            x0 = x1 = x2 = -b_a_3
+            return torch.stack([x2, x1, x0])
+
+        else:
+            cube_root = torch.pow(2 * R, 1 / 3)
+            x0 = cube_root - b_a_3
+            return x0
+
+    if D <= 0:
+        # three real roots
+        theta = torch.acos(R / torch.sqrt(-Q3))
+        sqrt_Q = torch.sqrt(-Q)
+        x0 = 2 * sqrt_Q * torch.cos(theta / 3.0) - b_a_3
+        x1 = 2 * sqrt_Q * torch.cos((theta + 2 * _PI) / 3.0) - b_a_3
+        x2 = 2 * sqrt_Q * torch.cos((theta + 4 * _PI) / 3.0) - b_a_3
+        return torch.stack([x2, x1, x0])
+
+    # D > 0, only one real root
+    AD = torch.tensor(0, device=a.device, dtype=a.dtype)
+    BD = torch.tensor(0, device=a.device, dtype=a.dtype)
+    R_abs = torch.abs(R)
+    if R_abs > 1e-16:
+        AD = torch.pow(R_abs + torch.sqrt(D), 1 / 3)
+        AD = AD if R >= 0 else -AD
+        BD = -Q / AD
+    x0 = AD + BD - b_a_3
+
+    return x0
+
+
+# Reference: Adapted from the 'run_7point' function in opencv
+# https://github.com/opencv/opencv/blob/4.x/modules/calib3d/src/fundam.cpp
+def run_7point(points1: torch.Tensor, points2: torch.Tensor) -> torch.Tensor:
+    r"""Compute the fundamental matrix using the 7-point algorithm.
+
+    Args:
+        points1: A set of points in the first image with a tensor shape :math:`(B, N, 2), N==7`.
+        points2: A set of points in the second image with a tensor shape :math:`(B, N, 2), N==7`.
+
+    Returns:
+        the computed fundamental matrix with shape :math:`(B, 3*m, 3), Valid values of m are 1, 2 or 3`
+    """
+    if points1.shape != points2.shape:
+        raise AssertionError(points1.shape, points2.shape)
+    if points1.shape[1] != 7:
+        raise AssertionError(points1.shape)
+
+    points1_norm, transform1 = normalize_points(points1)
+    points2_norm, transform2 = normalize_points(points2)
+
+    x1, y1 = torch.chunk(points1_norm, dim=-1, chunks=2)  # Bx1xN
+    x2, y2 = torch.chunk(points2_norm, dim=-1, chunks=2)  # Bx1xN
+
+    ones = torch.ones_like(x1)
+    # form a linear system: which represents
+    # the equation (x2[i], 1)*F*(x1[i], 1) = 0
+    X = torch.cat([x2 * x1, x2 * y1, x2, y2 * x1, y2 * y1, y2, x1, y1, ones], dim=-1)  # BxNx9
+
+    # X * Fmat = 0 is singular (7 equations for 9 variables)
+    # solving for nullspace of X to get two F
+    _, _, v = _torch_svd_cast(X)
+
+    # last two singluar vector as a basic of the space
+    f1 = v[0, :, 7]
+    f2 = v[0, :, 8]
+
+    # lambda*f1 + mu*f2 is an arbitary fundamental matrix
+    # f ~ lamda*f1 + (1 - lambda)*f2
+    # det(f) = det(lambda*f1 + (1-lambda)*f2), find lambda
+    # form a cubic equation
+    # finding the coefficients of cubic polynomial (coeffs)
+
+    coeffs = torch.zeros(4, device=v.device, dtype=v.dtype)
+
+    t0 = f2[4] * f2[8] - f2[5] * f2[7]
+    t1 = f2[3] * f2[8] - f2[5] * f2[6]
+    t2 = f2[3] * f2[7] - f2[4] * f2[6]
+
+    coeffs[3] += f2[0] * t0 - f2[1] * t1 + f2[2] * t2
```

Comment:
Can you add a torch compile test too? I think it does not allow in-place op

ex https://github.com/kornia/kornia/blob/master/test/filters/test_median.py#L101